taipy/core/_repository/_sql_repository.py

```diff
@@ -205,10 +211,20 @@ def __insert_model(self, model: ModelType):
         self.db.execute(str(query.compile(dialect=sqlite.dialect())), model.to_list())
         self.db.commit()
 
-    def _update_entry(self, model):
+    def __update_entry(self, model):
         query = self.table.update().filter_by(id=model.id)
-        self.db.execute(str(query.compile(dialect=sqlite.dialect())), model.to_list() + [model.id])
+        cursor = self.db.execute(str(query.compile(dialect=sqlite.dialect())), model.to_list() + [model.id])
         self.db.commit()
+        cursor.close()
+
+    def _update_entry(self, model):
+        start = datetime.now()
```

Comment:
Should we use the same mechanism as we are having on FS repo?
Probably the same decorator

Reply:
i dont see why not :D

Reply:
I would not let the user define how long to wait. 
Why not for the retry nb, but it has to be for good reasons.

With `filesystem`, we can know if a file exists before trying to read it. So if the file exists and we cannot read it, we can raise an exception that is catch to retry.
With `sql`, we must read the database to know if the entity exists. So if reading the entity fails, we cannot know if it is because it does not exist (which is normal) or because of a technical issue like the locked database. We need to understand how to differentiate the 2 cases to implement the retry.

Reply:
I think for now, I'd prefer to have something simple to see if it fixes the issue on integration tests. the function for retry we have in FS is for read only and the discussion we're having is for write in sql. So to have something for it, we should introduce new term like the one using in read function `Config.core.read_entity_retry`, we should have something like `Config.core.write_entity_retry` for write also. What do you guys think?